pallets/bridge/src/lib.rs

```diff
@@ -810,10 +826,33 @@ impl<T: Config> Pallet<T> {
     }
 
     /// Cancels a proposal.
+    #[require_transactional]
     fn cancel_materialize(src_id: ChainId, src_nonce: ChainNonce) -> DispatchResult {
         Self::update_bridge_registry(src_id, src_nonce);
         Self::deposit_event(Event::ProposalRejected(src_id, src_nonce));
 
         Ok(())
     }
+
+    #[require_transactional]
+    fn gift_fees(who: T::AccountId, asset_id: CurrencyId, amount: BalanceOf<T>) -> DispatchResult {
```

Comment:
gift_fees looks like a getter but there are mutations inside

Reply:
> gift_fees looks like a getter but there are mutations inside

how is `grant_incentive_bonus`?

Reply:
yeah it works